### PR TITLE
map position between gazebo and vizkit3d

### PR DESCRIPTION
**Pull request by gustavoneves12**
add script to map position between gazebo and vizkit3d
add config file to setup links that will be mapping
create a data folder with resources files

### CODE REVIEW

File: scripts/start_vizkit3d_ocean.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    exit 0
+end
+
+if flat_fish_task.state  == :PRE_OPERATIONAL
+    flat_fish_task.configure
+end
+
+flat_fish_model = nil
+ 
+flat_fish_task.exported_links.each do |link_name|
```
Comment (doudou) [inline scripts/start_vizkit3d_ocean.rb]: This makes no sense. 
 - you iterate over a collection which means that you assume that there are more than one item
 - and then update the same model (which is correct), which means that you assume there will be only one link

You have two options there:
 - ensure that exported_links is of size 1 (and raise an exception if it does not)
 - hardcode the reference link name in the script (and make sure it is present in exported_links)

I would prefer the second option, as it is IMO more robust to changes in the gazebo configuration

File: config/orogen/rock_gazebo::ModelTask.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+--- name:default
+exported_links: ['flat_fish_body']
+
+# no documentation available for this property
+metadata:
```
Comment (doudou) [inline config/orogen/rock_gazebo::ModelTask.yml]: This is a bug in Rock. Remove the metadata property from here, it is not supposed to be stored in config files.

File: data/models/urdf/flatfish.urdf [hunk 3]
```diff
@@ -1,13 +1,10 @@
...
-	<link name="base_link">
-	    
+<robot name="flat_fish">
+	<link name="flatfish_body">
+	   <origin xyz="0.0 0 0" rpy="0.0 0.0 0.0"/>
 	    <visual>
-			<origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
 			<geometry>
-				<mesh filename="FLATFISH/flatfish_05.osg"/>
+				<mesh filename="../meshes/flat_fish/visual.osg"/>
```
Comment (doudou) [inline data/models/urdf/flatfish.urdf]: Be consistent in your namings. visual.osg for flat_fish but manifold.osg for the manifold.... (I prefer the latter)

File: scripts/start_gazebo.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+ENV['GAZEBO_MODEL_PATH']="#{ENV['AUTOPROJ_CURRENT_ROOT']}/bundles/flat_fish/data/gazebo/models"
+puts `rock-gazebo $AUTOPROJ_CURRENT_ROOT/bundles/flat_fish/data/gazebo/worlds/underwater.world`
```
Comment (doudou) [inline scripts/start_gazebo.rb]: This does nothing ;-)

File: scripts/start_vizkit3d_ocean.rb [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+manifold_model.loadModel "#{ENV['AUTOPROJ_CURRENT_ROOT']}/bundles/flat_fish/data/models/meshes/manifold/manifold.osg"
+manifold_model.setPose(Qt::Vector3D.new(0, 0,-20),Qt::Quaternion::fromAxisAndAngle(0,0,0,0))
+
+dock_station = Vizkit.default_loader.RigidBodyStateVisualization
+dock_station.loadModel "#{ENV['AUTOPROJ_CURRENT_ROOT']}/bundles/flat_fish/data/models/meshes/dock_station/dock_station.osg"
+dock_station.setPose(Qt::Vector3D.new(0, -3.0,-18.5),Qt::Quaternion::fromAxisAndAngle(0,0,0,0))
+
+flat_fish_model = Vizkit.default_loader.RobotVisualization
+flat_fish_model.modelFile = "#{ENV['AUTOPROJ_CURRENT_ROOT']}/bundles/flat_fish/data/models/urdf/flatfish.urdf"
+flat_fish_model.setPose(Qt::Vector3D.new(0, 0, -15),Qt::Quaternion::fromAxisAndAngle(0,0,0,0))
```
Comment (doudou) [inline scripts/start_vizkit3d_ocean.rb]: > Qt::Quaternion::fromAxisAndAngle(0,0,0,0)

This constructs an identity quaternion. This is default in Qt (i.e. Qt::Quaternion.new does the same). I prefer the latter as it is hard to understand what the former does.

File: scripts/start_vizkit3d_ocean.rb (hunk 1 above)
Comment (gustavoneves12) [inline scripts/start_vizkit3d_ocean.rb]: I iterated over exported_links. If your model has more than one link, you need to update the position over all exported links. This code isn't finished. Currently the updateData change root node position. But we need to create an updateData function that changes a specific node position. For example, if we have a link to vehicle and another link to arm?

File: scripts/start_gazebo.rb (hunk 4 above)
Comment (gustavoneves12) [inline scripts/start_gazebo.rb]: This exports environment variable and runs rock-gazebo.

File: scripts/start_vizkit3d_ocean.rb (hunk 1 above)
Comment (doudou) [inline scripts/start_vizkit3d_ocean.rb]: > For example, if we have a link to vehicle and another link to arm?

Then you'll load multiple models and attach them each to their respective link. This is how SDF and URDF do it.

File: scripts/start_gazebo.rb (hunk 4 above)
Comment (doudou) [inline scripts/start_gazebo.rb]: Nope. It sets the envvar and displays the command line to rock-gazebo (There's a "puts" at the beginning)

Comment (gustavoneves12) [inline scripts/start_gazebo.rb]: It performs rock-gazebo too.

File: scripts/start_vizkit3d_ocean.rb (hunk 1 above)
Comment (gustavoneves12) [inline scripts/start_vizkit3d_ocean.rb]: Sorry, I don't understand. What do you mean with "load multiple models and attach them each to their respective link"? Will I load each link using RigidBodyStateVisualization for example? Sorry, but I didn't get it. Can you explain me with more details?

File: scripts/start_gazebo.rb (hunk 4 above)
Comment (doudou) [inline scripts/start_gazebo.rb]: My bad, missed the backquotes ... this would be a typical use for a shell script instead of ruby ...

**Comment by Thomio-Watanabe** [comment]
- change link port name 
- set flat_fish parameters and buoyancy variables

**Comment by doudou** [comment]
Hey. I've created #2 to add the SDF models to master, but I believe that we can drop the rest (we won't use the URDF models, and the scripts are obsolete)

**Pull request closed without merging**